### PR TITLE
Make sure window.lab points to the current JupyterLab instance.

### PR DESCRIPTION
**Pull request by afshin**

### CODE REVIEW

**Comment by jasongrout** [comment]
Following up on the JupyterLab Gitter conversation - I don't think we should expose a global like this. It encourages people to not use the plugin system to get the app as passed into their plugin. We can have a debug plugin that exposes this global if it's useful in debugging. @afshin also brought up that a browser extension could use this, but again, we could have a plugin that exposes whatever global is needed for a browser extension.

**Comment by jasongrout** [comment]
CC @sccolbert, @ellisonbg for their thoughts on having a global

**Comment by afshin** [comment]
After discussing with @jasongrout yesterday and @sccolbert and @blink1073 today, `window.lab` is only set if the page configuration denotes that the application is in dev mode.

File: jupyterlab/staging/index.js [hunk 1]
```diff
@@ -149,14 +149,19 @@ function main() {
...
+  var lab = new JupyterLab({
     mimeExtensions: mimeExtensions,
     disabled: disabled,
     deferred: deferred
   });
   register.forEach(function(item) { lab.registerPluginModule(item); });
   lab.start({ ignorePlugins: ignorePlugins });
 
+  // Expose global lab instance when in dev mode.
+  if ((PageConfig.getOption('devMode') || '').toLowerCase() === 'true') {
```
Comment (ian-r-rose) [inline jupyterlab/staging/index.js]: Will this ever evaluate to `true`? Is the fact that there are two `index.js`s enough?

Comment (afshin) [reply jupyterlab/staging/index.js]: If you start with `jupyter lab --dev-mode` that expression will evaluate to `true`.

I am not sure I understand what you mean by the two `index.js`s.

Comment (afshin) [reply jupyterlab/staging/index.js]: (Basically when you have the red banner.)

Comment (ian-r-rose) [reply jupyterlab/staging/index.js]: I mean the `index.js` in `jupyterlab/staging` vs the `index.js` in `dev_mode`.

Comment (blink1073) [reply jupyterlab/staging/index.js]: The Dev mode version is copied to staging during the release process.  They can differ before that if needed.

Comment (ian-r-rose) [reply jupyterlab/staging/index.js]: Ah, got it. Thanks!

**Comment by ian-r-rose** [comment]
Thanks!

**Pull request merged**